docs/developer-manual/consume.md

```diff
@@ -4,31 +4,228 @@ sidebar_position: 1
+
+In the case of the FSP, the state is simply the FS script hash.
+The FSP exists to allow Orcfax to update the FS script while allowing integrators to launch dapps
+without having to role their own update mechanism.
+Orcfax reserves the ability to be able to say, migrate to Plutus V3 without impacting integrators.
+By first finding the FSP in the reference inputs of a script context,
+a validator can recognise the correct FS script hash.
+
+The C script represents the constitution.
+It holds the current valid pubkey that is permitted to sign statements.
+The FS script checks that every statement published has a valid signature with respect to this pubkey.
+The constitution is not utilised by integrators, and plays no further part in this document.
+
+## Details
+
+### Steps
+
+To integrate an orcfax statement safely in a plutus script, the script must perform the following steps:
+
+1. Verify FSP UTXO from reference inputs.
+Extract FS script hash from inline datum.
+2. Verify FS UTXO. Extract the inline datum and parse as a bytearray.
+        The value is the FS script hash, which determines the FS token
+3. From reference inputs, find the input(s) containing an FS token.
+        These are the FS inputs
+4. For each FS input, extract the inline datum and parse.
+        The datum will be of type `FsDat<t>` (described below).
+5. Extract the statement from the datum.
+6. Verify that the statement’s feed_id is the one required for the integration.
+7. Verify that the statement’s created_at timestamp is within the timeframe established by your specific business rules.
+
+### Verify FSP UTXO
+
+The FSP script hash must first be ascertained from a given deployment.
+(Example deployments are listed below.)
+This will provide the policy ID part of an FSP asset class.
+
+The FSP UTXO will contain the FSP script token.
+The FSP script token has the following token name (expressed in base 16/ hex)
+```
+fsp_script=#"000de140"
+```
+This is derived from `cip-67` script NFT label.
+
+### Extract FS script hash
+
+All datums are inlined, including that of the FSP state.
+The FSP datum is simply
+```aiken
+pub type FspDat = ByteArray
+```
+The value is the FS script hash.
+When Orcfax upgrades the Orcfax-publish protocol, the value of the datum will be updated
+to reflect the new FS script, while the FSP script will remain the same.
+
+### Verify FS UTXOs
+
+An FS UTXO must contain an FS token.
+The FS token is the only asset class of the FS script, and has token name of the empty bytearray.
+
+Note: Integrators must verify an FS UTXO via an FS token.
+Any fool can put a UTXO at the FS script address with any data they like.
+
+### Coercing FS data
+
+An FS UTXO will have an inlined datum.
+The datum datatype is a parameterized datatype and is expressed in aiken lang as follows
+
+```aiken
+pub type Datum<t> {
+  statement : Statement<t>,
+  context : Context
+}
+
+/// Used by consumers
```

Comment:
integrators?

Reply:
My distinction is:

+ integrators => dapp devs
+ consumers => anyone involved in a tx that consumes a statement. 

So both would be correct but consumers are more general.

Reply:
I like this distinction